idaes/generic_models/unit_models/skeleton_model.py

```diff
@@ -0,0 +1,154 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+The basic functionality is to allow a user to add any model of choice that
+can interface with other IDAES unit models that leverage the ControlVolume
+blocks. This is done by letting the user create ports and populate with
+members that match with state variables defined in an IDAES unit model.
+Also, a default initialize method is provided with a flexibility to pass a
+callback from the user.
+"""
+
+from pyomo.network import Port
+from pyomo.common.config import ConfigValue, In
+
+from idaes.core.process_base import declare_process_block_class, \
+    ProcessBlockData
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util import get_solver
+import idaes.logger as idaeslog
+
+__author__ = "Jaffer Ghouse"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("SkeletonUnitModel")
+class SkeletonUnitModelData(ProcessBlockData):
+    """
+    This is the class for a skeleteon unit model. This will provide a user
+    with a generic skeleton to add a custom or a surrogate unit model
+    when controlvolumes are not required.
+    """
+
+    # Create Class ConfigBlock
+    CONFIG = ProcessBlockData.CONFIG()
+    CONFIG.declare("dynamic", ConfigValue(
+        default=False,
+        domain=In([False]),
+        description="Dynamic model flag",
+        doc="""Model does not handle dynamics but should be set by user."""))
```

Comment:
Not quite sure what this doc string means. The first part says it doesn't, but the second part says the user should set it?

Also, a related note, the absence of `has_holdup` means that `ProcessBlockData._setup_dynamics` will fail (as it expects `has_holdup` to be present). I do not think this is a big problem here (at least not yet), but we should be a ware of this at least. We might even want to consider overloading `_setup_dynamics` with a `NotSupported` error.

Reply:
Updated the doc string to better explain what I meant. 

Regarding the second comment, actually it fails without the dynamic flag.